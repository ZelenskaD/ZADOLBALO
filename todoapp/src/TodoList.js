import React, {useState} from "react";
import { v4 as uuidv4 } from 'uuid';
import Todo from './Todo';
import NewTodoForm from './NewTodoForm';

const TodoList = () => {
    const INITIAL_STATE = [
        {id: uuidv4(), task: "Feed puppies"},
        {id: uuidv4(), task: "Make a dinner"}
    ]

    const [todos, setTodos] = useState(INITIAL_STATE);

    const addTodo = (newTodo) => {
        setTodos(todos =>
            [...todos,
                {id: uuidv4(), task: newTodo.task }

            ])
    }
    const removeTodo = (id) => {
        setTodos(todos.filter(todo => todo.id !== id))
    }

    return(
        <div>
            <h3> Todo List</h3>
            <NewTodoForm addTodo={addTodo}/>
            <div>
                {todos.map(({id, task}) => < Todo key={id}
                                                     id={id}
                                                     task={task}
                                                     removeTodo={removeTodo}/>)}
            </div>
        </div>
    )
}


export default TodoList;