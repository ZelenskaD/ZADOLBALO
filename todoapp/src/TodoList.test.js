import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import TodoList from "./TodoList";



//smoke test
it("render without crashing", function  ()  {
    render(<TodoList />)
});


//snapshot test
it("matches snapshot", function () {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add a new todo",  function () {
    const { getByLabelText, getByText, debug } = render(<TodoList />);
    // eslint-disable-next-line testing-library/no-debugging-utils
    debug();

    const input = screen.getByLabelText("Task:");
    const btn = screen.getByText("Add Todo");


    fireEvent.change(input, { target: { value: "Feed dogs please" } });

    fireEvent.click(btn);

    // eslint-disable-next-line testing-library/no-debugging-utils
    debug();

    expect(screen.getByText("Feed dogs please")).toBeInTheDocument();
});

it("should remove a todo", () => {
    render(<TodoList />);

    // Initial todo items
    const initialTodos = ["Feed puppies", "Make a dinner"];

    // Verify that initial todos are in the document
    initialTodos.forEach(todo => {
        expect(screen.getByText(todo)).toBeInTheDocument();
    });

    // Remove the first todo
    const removeButtons = screen.getAllByText("X"); // Select all remove buttons
    fireEvent.click(removeButtons[0]); // Click the first remove button

    // Check if the first todo is removed
    expect(screen.queryByText("Feed puppies")).not.toBeInTheDocument();

    // Check if the second todo is still present
    expect(screen.getByText("Make a dinner")).toBeInTheDocument();
});