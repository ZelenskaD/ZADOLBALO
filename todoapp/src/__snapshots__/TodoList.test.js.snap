// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
       Todo List
    </h3>
    <form>
      <label
        for="task"
      >
        Task:
      </label>
      <input
        id="task"
        name="task"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <div>
      <div>
        <ul>
          <li>
             Feed puppies
          </li>
          <button>
            X
          </button>
        </ul>
      </div>
      <div>
        <ul>
          <li>
             Make a dinner
          </li>
          <button>
            X
          </button>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
